calcom/cal.com
# fix: Timing  format related bugs fixed and booking page dates made consistent for recurring events

## What does this PR do?
When we are taking availability as input from the user , we are taking as 00:00 format but while showing/ rendering at some places, we are showing as 24 which looks too odd. In some places, it is showing 00 while at some places it is showing time as 24 .This pr resolves the issue and now user will be seeing only 00:min and not 24:min . 

Also there is a issue on booking page which we get after booking is confirmed/rescheduled or cancelled. When we go for recurring events more than 4 , lets we have chosen for 12 recuring, then it will show the rest of 8 as last recurring day which is wrong.

Fixes / Changes Made:-
1. Changed the time format from 24 to 00 and also the recurring dates inconsistency removed.
![Untitled (3)](https://github.com/calcom/cal.com/assets/66114276/852cf4f1-8818-469b-be79-00e18de90090)
After making changes:- 
![Untitled (5)](https://github.com/calcom/cal.com/assets/66114276/c7873417-ef49-494c-95cf-9bf2173999ac)
2. Availability Page before making changes
![Untitled (6)](https://github.com/calcom/cal.com/assets/66114276/92252c9c-a51a-4108-9f50-ae6ac16437c4)
After making changes
![Untitled (7)](https://github.com/calcom/cal.com/assets/66114276/160e1187-6b4f-44a8-b16c-dc1d583ce5b8)


3. go to event types ⇒ select an event ⇒ click on edit ⇒  click on availability →
 now one will see timing like this 24 hr while we are setting availability we are setting as 00:00
![Untitled (8)](https://github.com/calcom/cal.com/assets/66114276/f374a59a-3ef5-468c-b96a-5407116a519e)

after making changes or after editing EventAvailabilityTab which is present in eventtype, we get
![Untitled (9)](https://github.com/calcom/cal.com/assets/66114276/3d04e176-36b1-4e94-9357-abf15bb63766)

Before making changes in [schedule.tsx] inside web pages availability,  we see

![Untitled (10)](https://github.com/calcom/cal.com/assets/66114276/f96c060e-9ae4-4453-93fe-de18716821f8)

after making changes
![Untitled (11)](https://github.com/calcom/cal.com/assets/66114276/3d3d1282-1ab7-4232-b8c6-a1c248f76be5)

before making changes in  packages/features/schedules/components/ScheduleListItem.tsx we see 
![Untitled (12)](https://github.com/calcom/cal.com/assets/66114276/4345b19e-91b7-48ca-8c8d-daa6d518d94c)

after making changes
![Untitled (13)](https://github.com/calcom/cal.com/assets/66114276/b4d2db4a-7126-479a-b5b1-b5a11e455ffe)

before editing  /packages/features/schedules/components/DateOverrideList.tsx.   (date overrides ) 
![Untitled (14)](https://github.com/calcom/cal.com/assets/66114276/3e7d78d5-03df-48fe-93a9-8fab5f7a9d54)

after making changes
![Untitled (15)](https://github.com/calcom/cal.com/assets/66114276/8360ffb7-2b7e-438c-ba2c-d814170767f3)
 
before applying changes , if meeting is scheduled/ rescheduled/ cancelled
![Untitled (16)](https://github.com/calcom/cal.com/assets/66114276/a08d5220-a220-4dc7-9d57-a00d910e5bab)

after applying changes -
<img width="1440" alt="Screenshot 2023-08-09 at 1 25 21 PM" src="https://github.com/calcom/cal.com/assets/66114276/a40dfad6-f9d2-4f7f-a824-68341638f60e">


## Type of change



## How should this be tested?
- create a recurring event  with more than 4 occurences , after successfully registering , expand the event day details and one will see the inconsistency. Also if we have taken time as 00:followed_by_mins then we will be seeing as 24:followed_by_mins.
- similarly while booking event , when we select slot we will select slot like 00 in case of 24 hr format but after selcting slot and before confirming ,we will see the slot at 24 which is not expected.This happens only in case of non recurring events.
- also while creating a schedule for a event type or adding overrides for a particular day, we will be entering as 00 but after adding,it will be shown to user as 24 instead of 00.
